src/iarc7_motion/iarc_tasks/go_to_roomba_task.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python
+
+import math
+import rospy
+import tf2_ros
+import tf2_geometry_msgs
+import threading
+
+from geometry_msgs.msg import TwistStamped
+from geometry_msgs.msg import PointStamped
+from geometry_msgs.msg import Point
+from iarc7_msgs.msg import OdometryArray
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Vector3Stamped
+from geometry_msgs.msg import Vector3
+
+from .abstract_task import AbstractTask
+from iarc_tasks.task_states import (TaskRunning,
+                                    TaskDone,
+                                    TaskCanceled,
+                                    TaskAborted,
+                                    TaskFailed)
+from iarc_tasks.task_commands import (VelocityCommand,
+                                      ArmCommand,
+                                      NopCommand)
+
+from height_holder import HeightHolder
+from height_settings_checker import HeightSettingsChecker
+from acceleration_limiter import AccelerationLimiter
+
+class GoToObjectTaskState(object):
+    init = 0
+    going = 1
+    waiting = 2
+
+class GoToRoombaTask(object, AbstractTask):
+
+    def __init__(self, task_request):
+
+        self._roomba_id = task_request.frame_id  + '/base_link'
+        # true means hit roomba, false means block
+        self._ending_radius_to_roomba = task_request.ending_radius_to_roomba
+
+        if self._roomba_id is None:
+            raise ValueError('A null roomba id was provided')
+
+        self._roomba_odometry = None
+        self._roomba_array = None
+        self._roomba_point = None
+        self._roomba_found = False
+        self._current_velocity = None
+
+        self._drone_odometry = None
+        self._canceled = False
+
+        self._tf_buffer = tf2_ros.Buffer()
+        self._tf_listener = tf2_ros.TransformListener(self._tf_buffer)
+
+        self._lock = threading.RLock()
+
+        self._roomba_status_sub = rospy.Subscriber(
+            'roombas', OdometryArray,
+            self._receive_roomba_status)
+
+        self._current_velocity_sub = rospy.Subscriber(
+            '/odometry/filtered', Odometry,
+            self._current_velocity_callback)
+
+        try:
+            self._TRANSFORM_TIMEOUT = rospy.get_param('~transform_timeout')
+            self._MAX_HORIZ_SPEED = rospy.get_param('~max_translation_speed')
+            self._MAX_START_TASK_DIST = rospy.get_param('~roomba_max_start_task_dist')
+            self._MAX_END_TASK_DIST = rospy.get_param('~roomba_max_end_task_dist')
+            self._MAX_Z_VELOCITY = rospy.get_param('~max_z_velocity')
+            self._K_X = rospy.get_param('~k_term_tracking_x')
+            self._K_Y = rospy.get_param('~k_term_tracking_y')
+            GO_TO_HEIGHT = rospy.get_param('~track_roomba_height')
+            _roomba_diameter = rospy.get_param('~roomba_diameter')
```

Comment:
You don't need these variables, for any overshoot you may employ, which I am thinking we do not have any overshoot at all, will not be dependent on roomba nor drone diameters.

Reply:
These are from Track Roomba Task and used in several places. There is another variable called self._overshoot. I thought _K_X and the rest are for PID control?

Reply:
So the idea of this task was to use the `TranslateStopPlanner` and not use any weird form of PID control that we use elsewhere, as it is not suitable for long distance translations.